### PR TITLE
Update Read/Write File to Operate more like Node's FS

### LINKED ISSUE #37: Encoding issue for readFile , writeFile works fine.
var AWS_ACCESS_KEY = 'xxxxxx';
var AWS_SECRET_KEY = 'yyyyyy';
var fs = require('fs');
var S3FS = require('s3fs');
var s3fsImpl = new S3FS("mybucket", {accessKeyId:AWS_ACCESS_KEY,secretAccessKey:AWS_SECRET_KEY});

module.exports.saveCode = function (req, res) {
  s3fsImpl.writeFile('sample_code.py', req.body.codeContent,  {encoding : 'utf8'}, function (err) {
    if (err) throw err;
  });
}

module.exports.fetchCode = function (req, res) {
  s3fsImpl.readFile('sample_code.py',  {encoding : 'utf8'}, function (err, data) {
    if (err) throw err;
    console.log(data);//Here is the issue, it's outputing <Buffer 74 69 74 69 20 3d 20 74 6f 74 6f 0a 74 61 74 61 20 3d 20 74 75 74 75>
  });
}
// In the 2nd argument I tried both  {encoding : 'utf8'} as well as just 'utf8'

### PR DESCRIPTION
**Pull request by DavidTPate**
Add functionality and tests around how we handle support for encoding in `readFile()` and `writeFile()` so that we are more in-line with how Node's FS works.

This way we are more in-line with how FS works on Node currently. Fixed #37

### CODE REVIEW

**Comment by DavidTPate** [comment]
@Jhorlin can you give this a once-over please?

**Comment by DavidTPate** [comment]
@RiptideElements/platform could someone review this please?

File: lib/s3fs.js [hunk 1]
```diff
@@ -941,6 +952,20 @@
...
 
+    function handleReadWriteFileOptions(options) {
+        if (typeof options === 'function' || !options) {
+            options = { encoding: null };
+        } else if (typeof options === 'string') {
+            options = { encoding: options };
+        } else if (!options) {
+            options = { encoding: null };
+        } else if (typeof options !== 'object') {
+            throw new TypeError('Bad arguments');
```
Comment (toaster33) [inline lib/s3fs.js]: would we get here?

File: lib/s3fs.js [hunk 2]
```diff
@@ -941,6 +952,20 @@
         });
     }
 
+    function handleReadWriteFileOptions(options) {
+        if (typeof options === 'function' || !options) {
```
Comment (toaster33) [inline lib/s3fs.js]: there a test for this?

File: lib/s3fs.js [hunk 3]
```diff
@@ -941,6 +952,20 @@
...
+            options = { encoding: null };
+        } else if (typeof options === 'string') {
+            options = { encoding: options };
+        } else if (!options) {
+            options = { encoding: null };
```
Comment (vbudhram) [inline lib/s3fs.js]: Don't need extra !options line since it is in first if

File: lib/s3fs.js (hunk 2 above)
Comment (DavidTPate) [inline lib/s3fs.js]: Yeah, it gets hit when there aren't any options passed, but we have a callback. We have tests around that.

File: lib/s3fs.js (hunk 1 above)
Comment (DavidTPate) [inline lib/s3fs.js]: I'll add a test around this line.

File: lib/s3fs.js (hunk 3 above)
Comment (DavidTPate) [inline lib/s3fs.js]: I know it's redundant, but if this is re-used in other methods I want to make sure we catch it.

File: lib/s3fs.js (hunk 1 above)
Comment (DavidTPate) [inline lib/s3fs.js]: Added!

**Pull request merged**